### PR TITLE
Added the remoteHub field (for Git repository name) in the ChaosHub Frontend

### PR DESCRIPTION
**Pull request by JanhaviAlekar**
## Proposed changes
Screenshot from 2024-08-22 15-48-16
Screenshot from 2024-08-22 15-48-50

Summarize your changes here to communicate with the maintainers and make sure to put the link of that issue

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).
 #4842

### CODE REVIEW

File: chaoscenter/web/src/images/Bitbucket.svg [hunk 1]
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="utf-8"?><!-- Uploaded to: SVG Repo, www.svgrepo.com, Generator: SVG Repo Mixer Tools -->
+<svg width="800px" height="800px" viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg" fill="none"><path fill="#2684FF" fill-rule="evenodd" d="M1.403 2.15A.43.43 0 011.73 2l12.54.002a.43.43 0 01.424.496l-1.81 11.135a.43.43 0 01-.425.36H3.693a.585.585 0 01-.568-.478l-1.82-11.02a.425.425 0 01.098-.345zm5.203 7.814H9.41l.677-3.93H5.859l.747 3.93z" clip-rule="evenodd"/><path fill="url(#bitbucket-color-16__paint0_linear_707_135)" d="M14.122 6.033H10.1l-.67 3.931H6.604L3.317 13.86c.105.09.238.139.376.14h8.766a.43.43 0 00.425-.36l1.238-7.607z"/><defs><linearGradient id="bitbucket-color-16__paint0_linear_707_135" x1="11.544" x2="6.918" y1="4.676" y2="11.282" gradientUnits="userSpaceOnUse"><stop offset=".18" stop-color="#0052CC"/><stop offset="1" stop-color="#2684FF"/></linearGradient></defs></svg>
```
Comment (SahilKr24) [inline chaoscenter/web/src/images/Bitbucket.svg:2]: we can update height and width 32px for all svgs or just remove them if not required.

File: chaoscenter/web/src/views/AddHubModalWizard/AddHubModalWizard.tsx [hunk 2]
```diff
@@ -231,6 +236,22 @@ const GitConnectionStep: React.FC<
...
+                <FormInput.DropDown
+                  name="remoteHub"
+                  label={<Text font={{ variation: FontVariation.FORM_LABEL }}>Remote Hub</Text>}
+                  placeholder={getString('remoteHub')}
+                  items={[
+                    { label: 'GitHub', value: 'GitHub' },
+                    { label: 'Bitbucket', value: 'Bitbucket' },
+                    { label: 'Azure Repo', value: 'Azure Repo' },
+                    { label: 'GitLab', value: 'GitLab' },
+                    { label: 'Others', value: 'Others' }
```
Comment (SahilKr24) [inline chaoscenter/web/src/views/AddHubModalWizard/AddHubModalWizard.tsx:248]: strings

File: chaoscenter/web/src/views/ChaosHubs/ChaosHubs.tsx [hunk 3]
```diff
@@ -48,6 +52,25 @@ function ConnectionStatus({ isConnected }: { isConnected: boolean }): React.Reac
...
+      case 'GitLab':
+        return <img src={Gitlab} height={27.38} width={29} alt="Gitlab" />;
+      default:
+        return <img src={privateHubLogo} height={27.38} width={29} alt="Private Hub" />;
+    }
+  };
+
+  return hubSvg();
+};
+
```
Comment (SahilKr24) [inline chaoscenter/web/src/views/ChaosHubs/ChaosHubs.tsx:73]: instead of switching the `img` tag we can switch the `src` directly

Comment (SahilKr24) [reply chaoscenter/web/src/views/ChaosHubs/ChaosHubs.tsx:73]: we can shift this to utils

File: chaoscenter/web/src/views/AddHubModalWizard/AddHubModalWizard.tsx [hunk 4]
```diff
@@ -231,6 +236,22 @@ const GitConnectionStep: React.FC<
                   placeholder={getString('enterHubRepositoryBranch')}
                 />
 
+                <FormInput.DropDown
+                  name="remoteHub"
+                  label={<Text font={{ variation: FontVariation.FORM_LABEL }}>Remote Hub</Text>}
```
Comment (SahilKr24) [inline chaoscenter/web/src/views/AddHubModalWizard/AddHubModalWizard.tsx:241]: strings

File: chaoscenter/web/src/views/ChaosHubs/ChaosHubs.tsx [hunk 5]
```diff
@@ -14,8 +14,12 @@ import type { ListChaosHubRequest, ListChaosHubResponse, SyncChaosHubRequest, Sy
 import CustomTagsPopover from '@components/CustomTagsPopover';
 import { useDocumentTitle, useRouteWithBaseUrl } from '@hooks';
 import NoFilteredData from '@components/NoFilteredData';
+import GitHub from '@images/Github.svg';
+import Azure from '@images/Azure.svg';
+import Gitlab from '@images/Gitlab.svg';
 import enterpriseHubLogo from '../../images/enterpriseHub.svg';
 import privateHubLogo from '../../images/privateHub.svg';
+import Bitbucket from '../../images/Bitbucket.svg';
```
Comment (SahilKr24) [inline chaoscenter/web/src/views/ChaosHubs/ChaosHubs.tsx:22]: import all from `@images`

File: chaoscenter/web/src/views/ChaosHubs/ChaosHubs.tsx [hunk 6]
```diff
@@ -132,7 +155,7 @@ export const ChaosHubsView: React.FC<ChaosHubParams> = ({
                       {chaosHub.isDefault ? (
                         <img src={enterpriseHubLogo} height={26.85} width={29} alt="Enterprise Hub" />
                       ) : (
-                        <img src={privateHubLogo} height={27.38} width={29} alt="Private Hub" />
+                        <RemoteHubImage remoteHubName={chaosHub.remoteHub} />
```
Comment (SahilKr24) [inline chaoscenter/web/src/views/ChaosHubs/ChaosHubs.tsx:158]: we can pass `isDefault` too as a prop and get the correct icon

File: chaoscenter/web/src/views/EditHubModalWizard/EditHubModalWizard.tsx [hunk 7]
```diff
@@ -229,6 +232,22 @@ const GitConnectionStep: React.FC<
...
+                  name="remoteHub"
+                  label={<Text font={{ variation: FontVariation.FORM_LABEL }}>Remote Hub</Text>}
+                  placeholder={'Select Remote hub'}
+                  items={[
+                    { label: 'GitHub', value: 'GitHub' },
+                    { label: 'Bitbucket', value: 'Bitbucket' },
+                    { label: 'Azure Repo', value: 'Azure Repo' },
+                    { label: 'GitLab', value: 'GitLab' },
+                    { label: 'Others', value: 'Others' }
+                  ]}
```
Comment (SahilKr24) [inline chaoscenter/web/src/views/EditHubModalWizard/EditHubModalWizard.tsx:245]: strings

**Review by SahilKr24 — CHANGES_REQUESTED** [review]

**Review by SahilKr24 — APPROVED** [review]

**Review by amityt — APPROVED** [review]

**Pull request merged**